src/Textile/API/events.ts

```diff
@@ -0,0 +1,279 @@
+import {
+  DeviceEventEmitter,
+  NativeEventEmitter,
+  NativeModules,
+  Platform,
+} from 'react-native'
+
+import {
+  INotification,
+  Notification,
+  IFeedItem,
+  FeedItem,
+  IThread,
+  Thread,
+  IContact,
+  Contact,
+} from './model'
+
+const { TextileEvents } = NativeModules
+const eventEmitter = Platform.select({
+  android: DeviceEventEmitter,
+  ios: new NativeEventEmitter(TextileEvents),
+})
+
+let nodeStartedListeners: Array<() => void> = []
+let nodeFailedToStartListeners: Array<(error: string) => void> = []
+let nodeStoppedListeners: Array<() => void> = []
+let nodeFailedToStopListeners: Array<(error: string) => void> = []
+let nodeOnlineListeners: Array<() => void> = []
+let willStopNodeInBackgroundAfterDelayListeners: Array<(seconds: number) => void> = []
+let canceledPendingNodeStopListeners: Array<() => void> = []
+let notificationReceivedListeners: Array<(notification: INotification) => void> = []
+let threadUpdateReceivedListeners: Array<(feedItem: IFeedItem) => void> = []
+let threadAddedListeners: Array<(threadId: string) => void> = []
+let threadRemovedListeners: Array<(threadId: string) => void> = []
+let accountPeerAddedListeners: Array<(peerId: string) => void> = []
+let accountPeerRemovedListeners: Array<(peerId: string) => void> = []
+let queryDoneListeners: Array<(queryId: string) => void> = []
+let queryErrorListeners: Array<(queryId: string, error: string) => void> = []
+let clientThreadQueryResultListeners: Array<(queryId: string, thread: IThread) => void> = []
+let contactQueryResultListeners: Array<(queryId: string, contact: IContact) => void> = []
+
+export class EventSubscription {
```

Comment:
All the `addListener` methods return an instance of `EventSubscription` that the subscriber can hold onto and cancel at a later time.